cdap-proto/src/main/java/co/cask/cdap/proto/ProtoTrigger.java

```diff
@@ -245,22 +218,22 @@ public ProgramStatus getProgramStatus() {
 
     @Override
     public void validate() {
-      ProtoTrigger.validateNotNull(getNamespace(), "program namespace");
-      ProtoTrigger.validateNotNull(getApplication(), "program application");
-      ProtoTrigger.validateNotNull(getApplicationVersion(), "program application version");
-      ProtoTrigger.validateNotNull(getProgramName(), "program name");
-      ProtoTrigger.validateNotNull(getProgramType(), "program type");
+      ProtoTrigger.validateNotNull(getProgramId(), "program id");
+      ProtoTrigger.validateNotNull(getProgramId().getProgram(), "program name");
```

Comment:
the following validation is not necessary since `ProgramId` already performs such check internally